File: CS/PopupContainerFormOkButton.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.XtraEditors.Popup;
using DevExpress.XtraEditors;
using System.Windows.Forms;
using System.Drawing;

namespace PopupContainerControlWithOKButton
{
    class PopupContainerFormOkButton : PopupContainerForm
    {
        // constructor
        public PopupContainerFormOkButton(PopupContainerEditOKButton ownerEdit) : base(ownerEdit) {}

        // override methods
        protected override void SetupButtons()
        {
            UpdatePopupButtons();
        }

        new RepositoryItemPopupContainerEditOKButton Properties
        {
            get
            {
                PopupContainerEditOKButton edit = OwnerEdit as PopupContainerEditOKButton;
                if (edit == null) return null;
                return edit.Properties;
            }
        }
        internal void UpdatePopupButtons()
        {
            if (Properties == null) return;
            this.fShowOkButton = Properties.ShowOkButton;
            if (Properties.ShowPopupCloseButton)
                this.fCloseButtonStyle = Properties.ShowOkButton ? BlobCloseButtonStyle.Caption : BlobCloseButtonStyle.Glyph;
            else
                this.fCloseButtonStyle = BlobCloseButtonStyle.None;
            this.AllowSizing = Properties.PopupSizeable;
            if (!AllowSizing && !fShowOkButton && fCloseButtonStyle == BlobCloseButtonStyle.None)
                ViewInfo.ShowSizeBar = false;
        }
    }
}

File: VB/PopupContainerFormOkButton.vb
Imports DevExpress.XtraEditors.Popup
Imports System.Drawing

Namespace PopupContainerControlWithOKButton

    Friend Class PopupContainerFormOkButton
        Inherits PopupContainerForm

        ' constructor
        Public Sub New(ByVal ownerEdit As PopupContainerEditOKButton)
            MyBase.New(ownerEdit)
        End Sub

        ' override methods
        Protected Overrides Sub SetupButtons()
            UpdatePopupButtons()
        End Sub

        Private Overloads ReadOnly Property Properties As RepositoryItemPopupContainerEditOKButton
            Get
                Dim edit As PopupContainerEditOKButton = TryCast(OwnerEdit, PopupContainerEditOKButton)
                If edit Is Nothing Then Return Nothing
                Return edit.Properties
            End Get
        End Property

        Friend Sub UpdatePopupButtons()
            If Properties Is Nothing Then Return
            fShowOkButton = Properties.ShowOkButton
            If Properties.ShowPopupCloseButton Then
                fCloseButtonStyle = If(Properties.ShowOkButton, BlobCloseButtonStyle.Caption, BlobCloseButtonStyle.Glyph)
            Else
                fCloseButtonStyle = BlobCloseButtonStyle.None
            End If

            AllowSizing = Properties.PopupSizeable
            If Not AllowSizing AndAlso Not fShowOkButton AndAlso fCloseButtonStyle = BlobCloseButtonStyle.None Then ViewInfo.ShowSizeBar = False
        End Sub
    End Class
End Namespace
